server/src-lib/Hasura/Server/Utils.hs

```diff
@@ -227,3 +229,17 @@ makeReasonMessage errors showError =
     [singleError] -> "because " <> showError singleError
     _ -> "for the following reasons:\n" <> T.unlines
          (map (("  • " <>) . showError) errors)
+
+executeJSONPath :: JSONPath -> Value -> IResult Value
+executeJSONPath jsonPath = iparse (valueParser jsonPath)
+  where
+    valueParser path value = case path of
+      []                      -> pure value
+      [pathElement]           -> parseWithPathElement pathElement value
```

Comment:
This case match isn't required :smiley: . The next match handles this case too. Feel free to delete this line.

Reply:
If I omit the `[]` case, it will give a non-exhaustive pattern match warning, so that's why I have included it

Reply:
No No, I mean 
```haskell
 [pathElement]           -> parseWithPathElement pathElement value
```
this line.